domain/estimators/logistic.py

```diff
@@ -94,18 +93,23 @@ def _gen_training_data(self):
             assert(feat_tensor.shape[0] == len(domain_vals))
             self._X[sample_idx:sample_idx+len(domain_vals)] = feat_tensor
 
+            # If the init value is not NULL, then we want to use these possible value samples
+            # during training.
+            if rec['init_value'] != NULL_REPR:
+                self._train_idx[sample_idx:sample_idx + len(domain_vals)] = 1
+
             # Assign the tensor corresponding to the initial value with
-            # a target label of 1.
-            try:
+            # a target label of 1. If the initial value is NULL, we do not care because it will be ignored in the training anyways.
+            if rec['init_value'] != NULL_REPR:
```

Comment:
What if we just reverse the logic of this `if ` statement to:
```python
if rec['init_value'] == NULL_REPR:
    continue
````
Would we even need the `self._train_idx`? Can we just construct `self._X` and `self._Y` from the examples that do not have NULL `init_value`?

Reply:
It's because we want to infer on the cell during prediction since we construct the tensors just once initially and re-use it. 

This will need to change in the batching paradigm.

Reply:
ok... too complicated for now, ignore my comment